packages/chain-adapters/src/evm/utils.ts

```diff
@@ -12,3 +15,65 @@ export const getErc20Data = async (
   const { data: callData } = await erc20Contract.populateTransaction.transfer(to, value)
   return callData || ''
 }
+
+type CalcNetworkFeeCryptoBaseUnitArgs = FeeData & {
+  supportsEIP1559: boolean
+}
+
+export const calcNetworkFeeCryptoBaseUnit = (args: CalcNetworkFeeCryptoBaseUnitArgs) => {
+  const {
+    supportsEIP1559,
+    gasLimit,
+    gasPrice,
+    l1GasLimit,
+    l1GasPrice,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+  } = args
+
+  // l1 fee if exists or 0
+  const l1Fee = bnOrZero(l1GasPrice).times(bnOrZero(l1GasLimit))
+
+  // eip1559 fees
+  if (supportsEIP1559 && maxFeePerGas && maxPriorityFeePerGas) {
+    return bn(gasLimit).times(maxFeePerGas).plus(l1Fee).toFixed(0)
+  }
+
+  // legacy fees
+  return bn(gasLimit).times(gasPrice).plus(l1Fee).toFixed(0)
+}
+
+export type GetFeesArgs = {
+  adapter: EvmChainAdapter
+  data: string
+  to: string
+  value: string
+  from: string
+  supportsEIP1559: boolean
+}
+
+export type GetFeesReturn = Fees & {
```

Comment:
nitpick - instead of the `Return` type naming convention, we could keep as `Fees` and make the imported `Fees` -> `NetworkFees` to make a bit more distinction.

Reply:
This is a great suggestion, d0ca2bf3d7bab313df5d4286dde4563344b1b104